src/Array.mo

```diff
@@ -602,28 +609,62 @@ module {
   /// Space: O(1)
   public func values<T>(array : [T]) : Iter.Iter<T> = array.vals();
```

Comment:
do we also want 
```
public func entries<T>(array : [T]) : Iter.Iter<(Nat, T)> = array.entries()
```

Reply:
Good idea; I think we could reasonably do that. Do we want to include `entries()` in the compiler? Otherwise, we could maybe just implement this in the module itself.

Also, I wonder if it would make sense to call this `Array.enumerate()` for consistency with `Iter.enumerate()`. The precedent from other languages is that `entries()` applies when you have explicit key-value pairs whereas `enumerate()` has implicit indices for an ordered collection, although this situation is a bit of a grey area.